tools/ci_connector_ops/ci_connector_ops/pipelines/bases.py

```diff
@@ -115,19 +124,41 @@ async def run(self, *args, **kwargs) -> StepResult:
             StepResult: The step result following the step run.
         """
         self.started_at = datetime.utcnow()
+        if self.should_log:
```

Comment:
Idea: Can we go one step further and stub out the logger so we can get rid of the if for anywhere we log?

something like
```py
DISABLED_LOGGER = logging.getLogger()
DISABLED_LOGGER.disabled = True

Class Step:
    def __init__(self, context: PipelineContext) -> None:
        self.logger = self.context.logger if self.should_log else DISABLED_LOGGER
```

or if theres only certain things we want to log and not others then do it based on the log levels

```
'debug': logging.DEBUG,
'info': logging.INFO,
'warning': logging.WARNING,
'error': logging.ERROR,
'critical': logging.CRITICAL
```

Reply:
Love it